dashboard/app/models/school.rb

```diff
@@ -347,7 +360,7 @@ def self.merge_from_csv(filename, options = CSV_IMPORT_OPTIONS, write_updates =
         parsed = block_given? ? yield(row) : row.to_hash.symbolize_keys
         loaded = find_by_id(parsed[:id])
 
-        if loaded.nil?
+        if loaded.nil? && write_inserts
```

Comment:
I don't love this logically, if anyone has a cleaner way of doing the same thing. Basically, `loaded.nil?` specifies whether a school with a given ID exists already or not, and the write_inserts and write_updates options specify whether, for a given file, we want to write new rows / update existing rows (if any exist).

Reply:
I think the logic is ok, but perhaps some different names would make this a bit easier to read?  Some suggestions:
- write_update --> update_existing
- write_inserts --> insert_new
- parsed --> csv_entry / csv_record
- loaded --> db_entry / db_record / existing_entry

Just some suggestions, feel free to use or not use as you think best.

One more suggestion: the logic for merging with an existing entry (L376-417) is a bit long and it might help to split some or all of that logic into a separate function.

Reply:
Great ideas re: variable names, I will update those.

Definitely agree as well that logic has gotten very complicated for merging with an existing entry -- [this is what that method](https://github.com/code-dot-org/code-dot-org/blob/6fa564886869c4b4a6103c85319b8a015c08e960/dashboard/app/models/school.rb#L241) looked like 3 years ago, so it's become a bit of a frankenstein over the past couple years. I think it's maybe outside the scope of this PR to split into it's own function (and add better testing) here, but I can make a PLC tech debt note about this (I have a few other thoughts I need to write up about this import process anyway).